exp/controllers/awsmachinepool_controller.go

```diff
@@ -233,7 +234,47 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 		// that change will not trigger a refresh. Do not start an instance refresh if only userdata changed.
 		return asgsvc.CanStartASGInstanceRefresh(machinePoolScope)
 	}
+
+	// Find existing ASG
+	asg, err := r.findASG(machinePoolScope, asgsvc)
+	if err != nil {
+		conditions.MarkUnknown(machinePoolScope.AWSMachinePool, expinfrav1.ASGReadyCondition, expinfrav1.ASGNotFoundReason, err.Error())
```

Comment:
This seems a bit confusing to me: We set the condition as unknown because of ASGNotFoundReason reason. Reasons represent contractual semantics and so I'd expect ASGNotFoundReason semantic to represent strictly a not found in aws but here it means something else - any error on the request.
Then yet in l246 we are using it for the not found semantic.

Reply:
Yes, now that you point it out, that is confusing to me too. This is a bit of relocated code, but it's worth fixing. I updated this to `MarkFalse` and removed the event from the `asg == nil` branch, since that is normal on the first reconcile.